src/core/bonded_interactions/bonded_coulomb.hpp

```diff
@@ -44,34 +44,30 @@ int bonded_coulomb_set_params(int bond_type, double prefactor);
  *  @param[in]  q1q2      Product of the particle charges.
  *  @param[in]  iaparams  Interaction parameters.
  *  @param[in]  dx        %Distance between the particles.
- *  @param[out] force     Force.
- *  @retval false
  */
-inline bool calc_bonded_coulomb_pair_force(
-    double const q1q2, Bonded_ia_parameters const *const iaparams,
-    Utils::Vector3d const &dx, Utils::Vector3d &force) {
+inline boost::optional<Utils::Vector3d>
+calc_bonded_coulomb_pair_force(double const q1q2,
+                               Bonded_ia_parameters const *const iaparams,
```

Comment:
can't we pass the parameters by ```const &```?

Reply:
Yes, and this made the integration loop a lot more homogeneous. Only the cell system and the principal loop of files `*_inline.hpp` use Particle pointers now, because in these parts of the code we don't always know in advance if a particle exists.